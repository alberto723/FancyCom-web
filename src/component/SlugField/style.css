.screen {
  flex: 1;
  background-color: 'white';
  padding: 10px;
}
.container {
  display: flex;
  flex-direction: row;
  align-items: center;
}
.domainHeaderSection {
  background-color: grey;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: row;
  padding-left: 10px;
  padding-right: 5px;
  height: 40px;
  border-top-left-radius: 7px;
  border-bottom-left-radius: 7px;
}
.domainPrefixColor {
  color: white;
}
.domainPrefixSection {
  flex: 1;
  background-color: grey;
  height: '100%';
  flex-wrap: wrap;
}
.dropdownSection {
  background-color: grey;
  border-width: 1px;
  border-color: grey;
  border-style: solid;
  /* height: 40px; */
}
.dropdownControlSection {
  background-color: grey;
  border-width: 1px;
  border-color: grey;
  border-style: solid;
  color: white;
  height: 38px;
}
.slashSection {
  background-color: grey;
  padding-left: 5px;
  padding-right: 5px;
  height: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
}
.slugTextSection {
  display: flex;
  flex: 1;
  color: white;
  background-color: grey;
  height: 34px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-right-radius: 7px;
  border-bottom-right-radius: 7px;
  border-color: grey;
  border-style: solid;
}
.buttonsSection {
  display: flex;
  flex-direction: row;
  align-items: center;
}
.actionButton {
  padding-left: 5px;
  padding-right: 5px;
}
