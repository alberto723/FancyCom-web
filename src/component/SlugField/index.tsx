import FileCopyIcon from '@material-ui/icons/FileCopy';
import LaunchIcon from '@material-ui/icons/Launch';
import React, { useState } from 'react';
import Dropdown from 'react-dropdown';
import 'react-dropdown/style.css';
import { makeStyles } from '@material-ui/core/styles';
// import './style.css';

interface domainPrefixProps {
  key: string;
  label: string;
  value: string;
}

export interface SlugFieldProps {
  domainPrefix?: string;
  domainPrefixItems: domainPrefixProps[];
  showPrefixDropdown?: boolean;
  value?: string;
  showCopyButton?: boolean;
  showLinkIcon?: boolean;
  // style?: ViewStyle;
  placeholder?: string;
  label?: string;
}

export const SlugField: React.FC<SlugFieldProps> = ({
  domainPrefix,
  domainPrefixItems,
  showPrefixDropdown,
  value,
  showCopyButton,
  showLinkIcon,
  // style,
  placeholder,
  label
}) => {
  const styles: any = useStyles();
  const [slugText, setSlugText] = useState(value);
  const [domainPrefixText, setDomainPrefixText] = useState(domainPrefix);

  const onLinkCopyButton = () => {
    const urlValue = 'http://' + domainPrefixText + '/' + slugText;
    document.execCommand('copy');
    alert('Copied the url ' + urlValue);
  };

  const onLinkLaunchButton = () => {
    const urlValue = 'http://' + domainPrefixText + '/' + slugText;
    window.open(urlValue, '_black');
  };

  const _onSelect = (option: any) => {
    setDomainPrefixText(option.label);
  };

  return (
    <div className={styles.screen}>
      <div className={styles.container}>
        <div className={styles.domainHeaderSection}>
          <p className={styles.domainPrefixColor}>https://</p>
        </div>
        <div className={styles.domainPrefixSection}>
          <Dropdown
            className={styles.dropdownSection}
            controlClassName={styles.dropdownControlSection}
            options={domainPrefixItems}
            onChange={_onSelect}
            value={domainPrefixItems[0]}
            placeholder='Select an option'
          />
        </div>
        <div className={styles.slashSection}>
          <p className={styles.domainPrefixColor}>/</p>
        </div>
        <input
          type='text'
          className={styles.slugTextSection}
          value={slugText}
          onChange={(ev: React.ChangeEvent<HTMLInputElement>): void => {
            setSlugText(ev.target.value);
          }}
        />
        <div className={styles.buttonsSection}>
          {showCopyButton && (
            <div className={styles.actionButton} onClick={onLinkCopyButton}>
              <FileCopyIcon style={{ color: 'grey', fontSize: 25 }} />
            </div>
          )}
          {showLinkIcon && (
            <div className={styles.actionButton} onClick={onLinkLaunchButton}>
              <LaunchIcon style={{ color: 'grey', fontSize: 25 }} />
            </div>
          )}
        </div>
      </div>
    </div>
  );
};

SlugField.defaultProps = {
  showCopyButton: true,
  showLinkIcon: true
};

const useStyles = makeStyles(() => ({
  screen: {
    flex: 1,
    backgroundColor: 'white',
    padding: '10px',
  },
  container: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  },
  domainHeaderSection: {
    backgroundColor: 'grey',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    paddingLeft: '10px',
    paddingRight: '10px',
    height: '40px',
    borderTopLeftRadius: '7px',
    borderBottomLeftRadius: '7px',
  },
  domainPrefixColor: {
    color: 'white',
  },
  domainPrefixSection: {
    flex: 1,
    backgroundColor: 'grey',
    height: '100%',
    flexWrap: 'wrap',
  },
  dropdownSection: {
    backgroundColor: 'grey',
    borderWidth: '1px',
    borderColor: 'grey',
    borderStyle: 'solid',
  },
  dropdownControlSection: {
    backgroundColor: 'grey',
    borderWidth: '1px',
    borderColor: 'grey',
    borderStyle: 'solid',
    color: 'white',
    height: '38px',
  },
  slashSection: {
    backgroundColor: 'grey',
    paddingLeft: '5px',
    paddingRight: '5px',
    height: '40px',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
  },
  slugTextSection: {
    display: 'flex',
    flex: 1,
    color: 'white',
    backgroundColor: 'grey',
    height: '34px',
    paddingLeft: '5px',
    paddingRight: '5px',
    borderTopRightRadius: '7px',
    borderBottomRightRadius: '7px',
    borderColor: 'grey',
    borderStyle: 'solid',
  },
  buttonsSection: {
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
  }
}));
