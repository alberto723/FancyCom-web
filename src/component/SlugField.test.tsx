// Canvas.test.js
import { render } from '@testing-library/react';
import React from 'react';
import renderer from 'react-test-renderer';

import { SlugField } from './SlugField';
import BlockIcon from '@material-ui/icons/Block';
import BorderOuterIcon from '@material-ui/icons/BorderOuter';
import BorderClearIcon from '@material-ui/icons/BorderClear';

const domainPrefixItems = [
  { key: '1', label: 'www.linkedin.com', value: 'www.linkedin.com' },
  { key: '2', label: 'www.google.com', value: 'www.google.com' },
  { key: '3', label: 'www.facebook.com', value: 'www.facebook.com' }
];

test('Should be rendered in the dom', () => {
  const component = renderer.create(
    <SlugField
      domainPrefix='www.google.com'
      domainPrefixItems={domainPrefixItems}
      showPrefixDropdown={true}
      placeholder='My-slug-name'
      value=''
      showCopyButton={true}
      showLinkIcon={true}
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should show the domainPrefix value in the dom', () => {
  const { queryByLabelText, getByText, container } = render(
    <SlugField
      domainPrefix='www.google.com'
      domainPrefixItems={domainPrefixItems}
      showPrefixDropdown={true}
      placeholder='My-slug-name'
      value=''
      showCopyButton={true}
      showLinkIcon={true}
    />
  );

  expect(container.querySelector('.makeStyles-domainPrefixSection-5')?.innerHTML.includes('linkedin')).toBe(true);
});

test('Should not render a dropdown when the showPrefixDropdown prop is false', () => {
  const { queryByLabelText, getByText, container } = render(
    <SlugField
      domainPrefix='www.google.com'
      domainPrefixItems={domainPrefixItems}
      showPrefixDropdown={true}
      placeholder='My-slug-name'
      value=''
      showCopyButton={true}
      showLinkIcon={true}
    />
  );

  expect(container.querySelectorAll('.makeStyles-domainPrefixSection-5').length).toBe(1);
});

test('Should show a [copy] button if the showCopyButton prop is true', () => {
  const { queryByLabelText, getByText, container } = render(
    <SlugField
      domainPrefix='www.google.com'
      domainPrefixItems={domainPrefixItems}
      showPrefixDropdown={true}
      placeholder='My-slug-name'
      value=''
      showCopyButton={true}
      showLinkIcon={true}
    />
  );

  expect(container.querySelector('.makeStyles-buttonsSection-10')?.innerHTML != null).toBe(true);
});

test('Should link the chosen domain prefix and slug value if provided (link icon href should match prop)', () => {
  const { queryByLabelText, getByText, container } = render(
    <SlugField
      domainPrefix='www.google.com'
      domainPrefixItems={domainPrefixItems}
      showPrefixDropdown={true}
      placeholder='My-slug-name'
      value=''
      showCopyButton={true}
      showLinkIcon={true}
    />
  );

  expect(container.querySelector('.makeStyles-buttonsSection-10')?.innerHTML != null).toBe(true);
});

test('Should render the label prop in the dom if provided', () => {
  const { queryByLabelText, getByText, container } = render(
    <SlugField
      domainPrefix='www.google.com'
      domainPrefixItems={domainPrefixItems}
      showPrefixDropdown={true}
      placeholder='My-slug-name'
      value=''
      showCopyButton={true}
      showLinkIcon={true}
    />
  );

  expect(container.querySelector('.makeStyles-domainPrefixSection-5')?.innerHTML.includes('linkedin')).toBe(true);
});
