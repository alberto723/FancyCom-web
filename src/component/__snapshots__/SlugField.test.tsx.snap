// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be rendered in the dom 1`] = `
<div
  className="makeStyles-screen-1"
>
  <div
    className="makeStyles-container-2"
  >
    <div
      className="makeStyles-domainHeaderSection-3"
    >
      <p
        className="makeStyles-domainPrefixColor-4"
      >
        https://
      </p>
    </div>
    <div
      className="makeStyles-domainPrefixSection-5"
    >
      <div
        className="Dropdown-root makeStyles-dropdownSection-6"
      >
        <div
          aria-haspopup="listbox"
          className="Dropdown-control makeStyles-dropdownControlSection-7"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <div
            className="Dropdown-placeholder is-selected"
          >
            www.linkedin.com
          </div>
          <div
            className="Dropdown-arrow-wrapper"
          >
            <span
              className="Dropdown-arrow"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="makeStyles-slashSection-8"
    >
      <p
        className="makeStyles-domainPrefixColor-4"
      >
        /
      </p>
    </div>
    <input
      className="makeStyles-slugTextSection-9"
      onChange={[Function]}
      type="text"
      value=""
    />
    <div
      className="makeStyles-buttonsSection-10"
    >
      <div
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "color": "grey",
              "fontSize": 25,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4l6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
          />
        </svg>
      </div>
      <div
        onClick={[Function]}
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          style={
            Object {
              "color": "grey",
              "fontSize": 25,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 19H5V5h7V3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2v-7h-2v7zM14 3v2h3.59l-9.83 9.83 1.41 1.41L19 6.41V10h2V3h-7z"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
